sql/derived_tables/po_lines_locations.sql

```diff
@@ -1,40 +1,40 @@
-DROP TABLE IF EXISTS po_lines_locations;
+/* This query creates a local table for purchase order line locations including location quantity and type.
+It uses conditional statements to search for location ID and location name in the 
+purchase order line location and holdings location fields, displays whichever is not null,
+and indicates the source of the resulting location data (purchase order location or purchase
+order holding location) in the pol_location_source column. 
+*/
 
--- Create a local table for Purchase Order Line Locations including location quantity and type.
+DROP TABLE IF EXISTS po_lines_locations;
 CREATE TABLE po_lines_locations AS
-WITH pol_location AS (
-    SELECT
-        pol.id AS pol_id,
-        json_extract_path_text(locations.data, 'locationId') AS pol_loc_id,
-        json_extract_path_text(locations.data, 'quantity') AS pol_loc_qty,
-        json_extract_path_text(locations.data, 'quantityElectronic') AS pol_loc_qty_elec,
-        json_extract_path_text(locations.data, 'quantityPhysical') AS pol_loc_quant_phys
+
+SELECT
+	pol.id AS pol_id,
+	json_extract_path_text(locations.data, 'quantity') AS pol_location_qty,
+	json_extract_path_text(locations.data, 'quantityElectronic') AS pol_loc_qty_elec,
+	json_extract_path_text(locations.data, 'quantityPhysical') AS pol_loc_qty_phys,      
+    CASE WHEN json_extract_path_text(locations.data, 'locationId') IS NOT NULL
+    	THEN json_extract_path_text(locations.data, 'locationId') 
+		ELSE ih.permanent_location_id END AS pol_location_id,	
+	CASE WHEN (il.name) IS NOT NULL THEN il.name
```

Comment:
Slight style inconsistency here - using parentheses around the field name here, but not in line 16 or 21. Also, in other lines, `THEN` is placed on a new line.

Reply:
Thank you for catching these!